lib/realtime/server.ex

```diff
@@ -80,6 +82,12 @@ defmodule Realtime.Server do
     )
   end
 
+  @spec stop_time_updates_for_trip(Trip.id()) :: [StopTimeUpdate.t()]
+  @spec stop_time_updates_for_trip(Trip.id(), GenServer.server()) :: [StopTimeUpdate.t()]
```

Comment:
Starting with the ets table, we definitely want that because a GenServer would limit concurrent reads.

In terms of this file, it is currently serving two jobs, handling channel subscriptions and also storing realtime data. I can definitely see the benefit of splitting out the storage job to a new module (or two modules, separating vehicles from STUs—the difference between 1 vs 2 here feels like a pretty minor nuance to me). That feels like a big change that I'd prefer to do outside of this PR though if that makes sense to you.

Reply:
The reads all happen sequentially by the one process that calls `Vehicle.from_vehicle_position`. It doesn't need to be called in parallel by multiple connecting clients. Given that, would we still need the ets table for performance?

It would be a big change to split storing vehicles from the vehicles subscriptions, but I just mean splitting the new block waivers from the existing vehicles. `server.ex` would probably have 0 lines changed, and all the new block waiver code would go in a new genserver. Like instead of calling `Server.update({:block_waivers, ...}` and `Server.stop_time_updates_for_trip()`, you'd call a new `BlockWaiversServer.update(...)`.